### PR TITLE
feat: integrated with get-starknet-wallet

### PR DESCRIPTION
**Pull request by avimak**
### Description of the Changes

Enhancement:
I've created a new library that is compatible with the current `getStarknet()` API but is able to support more wallets on StarkNet.
This PR integrates **starkgate-frontend** with that lib - adding multiple StarkNet wallets support in **starkgate-frontend** - without changing anything but one import line, and a few icons/texts to make the connect-L2-wallet step 100% wallet-agnostic.

### Checklist

- [x] Changes have been done against master branch, and PR does not conflict
- [ ] New unit / functional tests have been added (whenever applicable)  **-> not-applicable**
- [x] Test are passing in local environment
- [ ] Docs have been updated **-> non-required** 
- [x] PR title is follow the [Conventional Commits](https://www.conventionalcommits.org/) convention: `<type>[optional scope]: <description>`, e.g: `fix: prevent racing of requests`

---
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starkgate-frontend/163)

### CODE REVIEW

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: src/components/UI/WalletButton/WalletButton.js [hunk 1]
```diff
@@ -36,10 +38,11 @@ export const WalletButton = ({account, logoPath, onClick}) => {
       colorText={colorWhite}
       height={0}
       icon={
-        <DynamicIcon
-          path={logoPath}
-          size={isMobile(breakpoint) ? WALLET_LOGO_SIZE_MOBILE : WALLET_LOGO_SIZE}
-        />
+        logoPath.startsWith('data:image') ? (
+          <img alt={''} height={logoSize} src={logoPath} width={logoSize} />
```
[the remark this answers is not in the document]
Comment (avimak) [reply src/components/UI/WalletButton/WalletButton.js:42]: wallet window objects provide their own icon/logo data; here we show it via `` element

File: src/providers/WalletsProvider/WalletsProvider.js [hunk 2]
```diff
@@ -42,7 +42,15 @@ export const WalletsProvider = ({children}) => {
...
+      const wallet = getStarknet();
+      const enabled = await wallet
+        .enable(!autoConnect && {showModal: true})
+        .then(address => !!address?.length);
+      if (enabled) {
+        walletConfig.name = wallet.name || walletConfig.name;
+        walletConfig.logoPath = wallet.icon || walletConfig.logoPath;
+      }
+
       setL2WalletConfig(walletConfig);
```
[the remark this answers is not in the document]
Comment (avimak) [reply src/providers/WalletsProvider/WalletsProvider.js:54]: these 2 lines are called inside a `if enabled` clause -
```
      const wallet = getStarknet();
      const enabled = await wallet
        .enable(!autoConnect && {showModal: true})
        .then(address => !!address?.length);
      if (enabled) { // --> `enabled === true` means the wallet is successfully connected
        walletConfig.name = wallet.name || walletConfig.name;
        walletConfig.logoPath = wallet.icon || walletConfig.logoPath;
      }
```

Comment (dan-ziv) [inline src/providers/WalletsProvider/WalletsProvider.js:54]:
```suggestion
      const wallet = getStarknet();
      const enabled = await wallet
        .enable(!autoConnect && {showModal: true})
        .then(address => !!address?.length);
      if (enabled) {
        walletConfig.name = wallet.name || walletConfig.name;
        walletConfig.logoPath = wallet.icon || walletConfig.logoPath;
        setL2WalletConfig(walletConfig);
        }
```

File: src/components/UI/WalletButton/WalletButton.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dan-ziv) [reply src/components/UI/WalletButton/WalletButton.js:42]: OK, can you please move this logic to a `renderWalletLogo` function so the JSX will be more readable?

[the remark this answers is not in the document]
Comment (avimak) [reply src/components/UI/WalletButton/WalletButton.js]: done :)

**Review by dan-ziv — APPROVED** [review]

**Pull request merged**